pythainlp/tag/pos_tag.py

```diff
@@ -196,28 +219,35 @@ def pos_tag_transformers(
         raise ImportError(
             "Not found transformers! Please install transformers by pip install transformers")
 
-    if not words:
+    if not sentence:
         return []
 
-    if engine == "wangchanberta-ud-thai-pud-upos":
-        model = AutoModelForTokenClassification.from_pretrained(
-            "Pavarissy/wangchanberta-ud-thai-pud-upos")
-        tokenizer = AutoTokenizer.from_pretrained("Pavarissy/wangchanberta-ud-thai-pud-upos")
-    elif engine == "mdeberta-v3-ud-thai-pud-upos":
-        model = AutoModelForTokenClassification.from_pretrained(
-            "Pavarissy/mdeberta-v3-ud-thai-pud-upos")
-        tokenizer = AutoTokenizer.from_pretrained("Pavarissy/mdeberta-v3-ud-thai-pud-upos")
-    elif engine == "bert-base-th-cased-blackboard":
-        model = AutoModelForTokenClassification.from_pretrained("lunarlist/pos_thai")
-        tokenizer = AutoTokenizer.from_pretrained("lunarlist/pos_thai")
+    _blackboard_support_engine = {
+        "bert" : "lunarlist/pos_thai",
+    }
+
+    _pud_support_engine = {
+        "wangchanberta" : "Pavarissy/wangchanberta-ud-thai-pud-upos",
+        "mdeberta" : "Pavarissy/mdeberta-v3-ud-thai-pud-upos",
+    }
```

Comment:
note that i've made a dictionary to match the model name with the input engine so that it would be easier to maintain internal code in the future.

Reply:
Nice 

I think in general we need to update/expand the naming convention to cover our function names and model names as well. Also to better enforce it.

Current naming convention for files
https://github.com/PyThaiNLP/pythainlp/issues/141

Reply:
@bact agreed !, setting convention will reduce much complexity whether in user side and contributor side as well. Do you need me to put this into the pythainlp discussion section so that we can inform other contributors and also newcomers in the future too? if you're ok, i will put it in rn (of course that i will put it in Thai language krub)

Reply:
Yes please. Very appreciated.